modules/ServerAPI/src/client/SaltBrowser.coffee

```diff
@@ -268,45 +283,53 @@ class SaltBrowserController extends Backbone.View
 			dryrun = true
 
 			$.ajax(
-				url: "/api/cmpdRegAdmin/salts/" + dryrun, 
+				url: "/api/cmpdRegAdmin/salts?dryrun=" + dryrun, 
 				type: 'POST', 
 				data: JSON.stringify(saltDict)
 				contentType: 'application/json'
 				dataType: 'json'
 				success: (result) => # Result Success Should Be JSON of Salt
-					@$('.bv_createSalt').hide()
-					@$('.bv_confirmCreateSalt').show()
-
-					# Need to Parse Result Values Into Form Fields 
-
-					@$('.bv_abbrevNameConfirm').val result.abbrev
-					@$('.bv_saltNameConfirm').val result.name
-					@$('.bv_saltFormulaConfirm').val result.formula
-					@$('.bv_saltWeightConfirm').val "#{result.molWeight}"
-					@$('.bv_saltChargeConfirm').val "#{result.charge}"
-
-					# Need to Render Preview of Structure 
-					requestJSON = {
-						"molStructure" : result.molStructure,
-						"height" : 200, 
-						"width" : 200, 
-						"format" : "png"
-					}
-
-					# Render Image
-					$.ajax(
-						type: 'POST'
-						url: "/api/chemStructure/renderMolStructureBase64"
-						contentType: 'application/json'
-						dataType: 'text'
-						data: JSON.stringify(requestJSON)
-						success: (base64ImagePNG) => # Rendered Image Should Be Displayed on Sever
-							pngSrc = "data:image/png;base64," + base64ImagePNG
-							pngImage = '<img src="' + pngSrc + '" />'
-							@$('.bv_saltStructConfirm').html pngImage 
-						error: (result) =>
-							console.log(result)
-							@$('.bv_saltStructConfirm').hide()
+					console.log(result)
+					console.log(result.length)
+					if result.length > 1 # If there's an error / warning and no salt 
```

Comment:
I'm curious about this. Are we returning a different "shape" of JSON depending on whether this save succeeded or not?
Not something that needs to change right now, but it might make more sense to inspect the status code (which would be the second argument to the `success` callback) to determine which branch of the logic to carry out.

Reply:
Essentially yes, there are different shapes. Part of the reason why I did this was because of using the cmpdreg entity routes and not wanting to mess around too much with status codes since i'm not sure if that would break something else in ACAS. Definitely something worth looking into.

Reply:
Thanks. I'm good leaving this as-is for the time being.